### ISSUE TITLE
Modal without Jetpack Navigation

### ISSUE THREAD

**Issue by naixx** [WalletConnect/WalletConnectKotlinV2#1473]
**Is your feature request related to a problem? Please describe.**
Currently, the whole SDK is tightly coupled with Jetpack navigation. For the apps that use alternative navigation libraries like Voyager or Decompose it is extremely frustrating to integrate SDK. Even if someone wants to use `Web3ModalComponent` there is additional transitive dependency that is not required by the developer.

**Describe the solution you'd like**
SDK should provide navigation agnostic option or a clear example of how to use it.

**Describe alternatives you've considered**
Currently I try to integrate `Web3ModalComponent` into existing app and I can't do that without jetpack navigation. Having two nav libs is not what you want.

**Additional context**
Ideally, you should have a KMP version with low number of dependencies

**Comment by Raenar4k** [comment]
Yeah. Jetpack navigation also introduces issues, when WC uses older version and your up has newer one. This already caused frustrating crashes before:
https://github.com/WalletConnect/WalletConnectKotlinV2/issues/1299#issuecomment-1932074260
